engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -1,12 +1,15 @@
 /** @jest-environment setup-polly-jest/jest-environment-node */
 
 import { QuerySourceSkolemized } from '@comunica/actor-context-preprocess-query-source-skolemize';
+import type { Bindings } from '@comunica/bindings-factory';
+import { BindingsFactory } from '@comunica/bindings-factory';
 import { KeysHttpWayback, KeysQuerySourceIdentify } from '@comunica/context-entries';
 import { BlankNodeScoped } from '@comunica/data-factory';
 import type { QueryBindings, QueryStringContext } from '@comunica/types';
 import type * as RDF from '@rdfjs/types';
 import arrayifyStream from 'arrayify-stream';
 import 'jest-rdf';
+import '@comunica/jest';
```

Comment:
I'm surprised this works correctly.
We deliberately excluded this package, as this file is also tested in browsers using karma.
But the CI seems to pass, so I guess it's fine...

To be sure, can you double-check this by running the browser tests locally? (see the CI file on how to do this)

Reply:
To be double sure, I can just copy the logic of toEqualBindingsStream and test it that way, your call?

Reply:
Well, if it works correctly in the browser environment, then I guess we can keep the import there.

Reply:
@maartyman Could you check locally if the browser tests run fine for you?
Because I'd love to merge this!

Reply:
Will do! I was recently checking the performance and saw that the difference between `for` and `foreach` is actually pretty negligible. As the linter prefers `for`, I will change this and post the benchmarking results. This is the reason I haven't finalized this pr yet...

Reply:
Sounds good!

The reason I'm asking is because I was doing some profiling with a flame graph, and the (old) merge method is a significant bottleneck for one of our WatDiv queries. When using your version, the flame graph shows a reduction of 50% for that part of the graph. So that's quite something.